### PR TITLE
Upgrade wgpu

### PR DESCRIPTION
**Pull request by DJMcNab**
Wgpu 0.9 has released, now under MIT/Apache 2.0 🎉 

~~We can't remove MPL-2.0 from the deny.toml, because wgpu_subscriber is still under MPL.~~

### CODE REVIEW

**Comment by DJMcNab** [comment]
I've also tried to enable validation again, without success

Naga still chokes on our shaders, with exceedingly unhelpful error messages.

**Comment by khyperia** [comment]
@VZout I vaguely recall there was a reason you used `wgpu-subscriber` instead of `env_logger` in https://github.com/EmbarkStudios/rust-gpu/pull/92, am I misremembering and this change is fine, or is there a reason?

**Comment by DJMcNab** [comment]
I think wgpu moved to just log in https://github.com/gfx-rs/wgpu/issues/1211#issue-806966118

File: examples/runners/wgpu/src/compute.rs [hunk 1]
```diff
@@ -113,8 +113,9 @@ pub async fn start_internal(
         label: Some("Timestamps buffer"),
         size: 16,
         usage: wgpu::BufferUsage::MAP_READ | wgpu::BufferUsage::COPY_DST,
-        mapped_at_creation: false,
+        mapped_at_creation: true,
     });
+    timestamp_buffer.unmap();
```
Comment (VZout) [inline examples/runners/wgpu/src/compute.rs:118]: Why map and then immediately unmap?

**Review by VZout — CHANGES_REQUESTED** [review]

**Comment by VZout** [comment]
> @VZout I vaguely recall there was a reason you used `wgpu-subscriber` instead of `env_logger` in #92, am I misremembering and this change is fine, or is there a reason?

The reason we used `wgpu-subscriber` was because back then the triangle example in the wgpu repo used `wgpu-subscriber`. I'm not aware of any other reason. I think env logger is better anyway. gives you pretty colors.

File: examples/runners/wgpu/src/compute.rs (hunk 1 above)
Comment (DJMcNab) [reply examples/runners/wgpu/src/compute.rs:118]: That's required to actually get values from the queries - there's a wgpu issue about it, which I can't find at the moment

Comment (VZout) [reply examples/runners/wgpu/src/compute.rs:118]: 🤦

Comment (DJMcNab) [reply examples/runners/wgpu/src/compute.rs:118]: https://github.com/gfx-rs/wgpu/issues/1374#issuecomment-839359307

**Review by VZout — APPROVED** [review]

**Review by khyperia — APPROVED** [review]

**Pull request merged**